### PR TITLE
[ARCTIC-979][Core]Use alter to modify partition location rather than delete and add the partitions in UpdateHiveFiles commit

### PR DESCRIPTION
**Pull request by hzluting**
## Why are the changes needed?
improve #979 

## Brief change log

  - *use alter to modify partition location rather than delete and add the partitions in UpdateHiveFiles commit*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by zhoujinsong — APPROVED** [review]
LGTM.

@baiyangtx do you have time to review this PR?

File: hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java [hunk 1]
```diff
@@ -321,7 +321,9 @@ private Map<StructLike, Partition> filterNewPartitionNonExists(
               "partition will be " +
               "delete and re-create with same location");
         } else {
-          partitions.put(entry.getKey(), entry.getValue());
+          // this partition is need to alter, rather than delete
+          partitionToAlter.put(entry.getKey(), entry.getValue());
+          partitionToDelete.remove(entry.getKey());
```
Comment (baiyangtx) [inline hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java:326]: need to update hive partition fields and properties.
includes lastAccessTime, transient_lastDdlTime, totalSize, numRows, numFiles.

**Review by baiyangtx — CHANGES_REQUESTED** [review]

File: hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java (hunk 1 above)
Comment (hzluting) [reply hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java:326]: > need to update hive partition fields and properties. includes lastAccessTime, transient_lastDdlTime, totalSize, numRows, numFiles.

Yes, the partition is already produced by the latest info.

Comment (zhoujinsong) [reply hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java:326]: As I know, the `partition` from the `partitionToCreate` map is created with a new status.

reference:https://github.com/NetEase/arctic/blob/master/hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java#L192

**Pull request merged**